conans/client/cmd/new.py

```diff
@@ -347,6 +349,8 @@ def cmd_new(ref, header=False, pure_c=False, test=False, exports_sources=False,
         raise ConanException("'template' is incompatible with 'header', "
                              "'sources', 'pure-c' and 'bare'")
 
+    defines = defines or []
+    defines = dict((n, v) for n, v in (d.split('=') for d in defines))
```

Comment:
Better pass a dict, don't need to parse the inputs so inside, better do it in command.py and pass a dict to the api.

Reply:
agree, I suppose API method should already accept the dict, so conversion should happen earlier, just after the command line parsing

Reply:
moved